lib/models/instantiate-addons.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+/**
+@module ember-cli
+*/
+
+const DAGMap = require('dag-map').default;
+const logger = require('heimdalljs-logger')('ember-cli:addons-factory');
+const heimdall = require('heimdalljs');
+const SilentError = require('silent-error');
+
+/**
+ * Create instances of a set of "child" addons for a parent addon or project.
+ *
+ * @method initializeAddons
+ * @param {Object} parent an Addon or Project that is the direct containing object of the list
+ *   of children defined in addonPackages.
+ * @param {Project} project the project that contains the parent (so either the addon's project
+ *   if parent is an addon, or the project itself if it is a project). It is possible when
+ *   constructing custom addon instances that the project will actually be undefined--various
+ *   addon tests do this, for example.
+ * @param {Object} a map of addon name (including scope) to an AddonInfo with the name, path and
+ *   'pkg' object for that addon's package.json). These are what is turned into addons.
+ */
+module.exports = (parent, project, addonPackages) => {
```

Comment:
nit-pick: As this doesn't intend to close over `this`, I would prefer if this is a named `function` statement